android-components/components/feature/addons/src/main/java/mozilla/components/feature/addons/amo/AMOAddonsProvider.kt

```diff
@@ -469,34 +468,33 @@ internal fun JSONObject.getCategories(): List<String> {
 }
 
 internal fun JSONObject.getPermissions(): List<String> {
-    val fileJson = getJSONObject("current_version")
-        .getSafeJSONArray("files")
-        .getJSONObject(0)
-
-    val permissionsJson = fileJson.getSafeJSONArray("permissions")
+    val permissionsJson = getFile()?.getSafeJSONArray("permissions") ?: JSONArray()
     return (0 until permissionsJson.length()).map { index ->
         permissionsJson.getString(index)
     }
 }
 
 internal fun JSONObject.getCurrentVersion(): String {
-    return optJSONObject("current_version")?.getSafeString("version") ?: ""
+    return getJSONObject("current_version").getSafeString("version")
 }
 
-internal fun JSONObject.getDownload(): JSONObject? {
-    return (
-        getJSONObject("current_version")
-            .optJSONArray("files")
-            ?.getJSONObject(0)
-        )
+internal fun JSONObject.getFile(): JSONObject? {
+    return getJSONObject("current_version")
+        .getSafeJSONArray("files")
+        .optJSONObject(0)
 }
 
 internal fun JSONObject.getDownloadId(): String {
-    return getSafeString("id")
+    return getFile()?.getSafeString("id").orEmpty()
+}
+
+internal fun JSONObject.getCurrentVersionCreated(): String {
+    // We want to return: `current_version.files[0].created`.
+    return getFile()?.getSafeString("created").orEmpty()
```

Comment:
Just to confirm, now what we are checking it's the date when the latest extension file was created, right?

Reply:
Yeah, so essentially we are doing what AMO did a few years ago: https://github.com/mozilla/addons-frontend/issues/8861. This is needed for the featured add-ons only as we will have the correct™️ date for _installed_ add-ons soon (thanks to #3937).

The previous date was the last update of _anything_ in AMO (for a given add-on).